include/dlaf/factorization/cholesky/mc.h

```diff
@@ -114,19 +151,26 @@ void Cholesky<Backend::MC, Device::CPU, T>::call_L(Matrix<T, Device::CPU>& mat_a
 
 template <class T>
 void Cholesky<Backend::MC, Device::CPU, T>::call_L(comm::CommunicatorGrid grid,
-                                                   Matrix<T, Device::CPU>& mat_a) {
+                                                   Matrix<T, Device::CPU>& mat_a, comm::MPIMech mech) {
+  using hpx::resource::pool_exists;
+  using hpx::threads::thread_priority;
   using ConstTileType = typename Matrix<T, Device::CPU>::ConstTileType;
+  using hpx::util::unwrapping;
 
   auto executor_hp = dlaf::getHpExecutor<Backend::MC>();
   auto executor_np = dlaf::getNpExecutor<Backend::MC>();
-  auto executor_mpi = dlaf::getMPIExecutor<Backend::MC>();
 
-  common::Pipeline<comm::CommunicatorGrid> mpi_task_chain(std::move(grid));
+  // Set up MPI executor pipelines
+  std::string mpi_pool = (hpx::resource::pool_exists("mpi")) ? "mpi" : "default";
```

Comment:
I comment here because it is somehow related.

I was wondering if passing the `MPIMech` to all algorithms is needed. In particular:
- is it possible to mix different `MPIMech`? If not, passing it to different algorithms would not make sense to me.
- I think one reason you have to pass it as algorithm argument is because you have to initialize the `Executor`, and here there might be the link with @msimberg comment: would it be possible to "centralize" `Executor`s definition and then re-use them as other (and as @msimberg was suggesting)?

Reply:
@albestro Yes, good point!

> is it possible to mix different MPIMech? If not, passing it to different algorithms would not make sense to me.

This is technically possible but I don't see why someone might one to do it.

I can make `MPIMech` initialized once and remove it from all algorithms. I will check how this can be integrated with Mikaels initialization work.